test/fixtures/rules/std_rule_cases/L016.yml

```diff
@@ -455,3 +455,79 @@ test_comment_move_mid_query:
   configs:
     core:
       max_line_length: 40
+
+test_fix_implicit_indent:
+  # Test for ImplicitIndent.
+  # The theoretical indent between WHERE and "a" is implicit.
+  fail_str: |
+    SELECT
+        CASE
+            WHEN longer_and_longer AND much_much_much_longer
+                THEN longer_and_longer AND much_much_much_longer
+            ELSE longer_and_longer AND much_much_much_longer
+        END as foobar,
+        CASE WHEN a THEN b END as bar
+    FROM foo
+    WHERE a_really_long_field AND a_nother_really_long_field
+    HAVING a_really_long_field AND a_nother_really_long_field
+  fix_str: |
+    SELECT
+        CASE
+            WHEN longer_and_longer
+                AND much_much_much_longer
+                THEN longer_and_longer
+                    AND much_much_much_longer
+            ELSE longer_and_longer
+                AND much_much_much_longer
+        END as foobar,
+        CASE WHEN a THEN b END as bar
+    FROM foo
+    WHERE a_really_long_field
+        AND a_nother_really_long_field
+    HAVING a_really_long_field
+        AND a_nother_really_long_field
+  configs:
+    core:
+      max_line_length: 45
+    indentation:
+      allow_implicit_indents: True
+
+test_fix_no_implicit_indent:
+  # Test explicitly preventing implicit indents.
+  fail_str: |
+    SELECT
```

Comment:
I notice the `fail_str` for this test is different from the one for `test_fix_implicit_indent`, e.g. the indentation of the first `THEN`. Is that intentional? I had assumed they'd be the same, in order to illustrate the different outcomes.

Reply:
Assuming the two are logically the same even despite the indentation differences, I notice the output is 15 lines vs 20 lines. In that sense, I like the test because a 33% increase in lines is a high price to pay! It definitely reminds me of some annoying things Black does occasionally. FWIW, I code on a notebook, often a 13" notebook. 🙂

Reply:
They are supposed to be the same. Good catch.

Now you can see why I like this feature. :)

Reply:
Me too!